### PR TITLE
[C-1638] Download collections and display offline

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Changes:
1. Download collection metadata along with tracks
2. Load all offline collections into state on startup
3. Properly render albums and playlists tab, and collection details page when offline
4. Populate collection page lineup to enable offline collection playback

### Feature Flags ###

`OFFLINE_MODE_ENABLED`

### CODE REVIEW

File: packages/mobile/src/components/offline-downloads/DownloadToggle.tsx [hunk 1]
```diff
@@ -24,9 +25,10 @@ export type TrackForDownload = {
 }
 
 type DownloadToggleProps = {
-  labelText?: string
   tracksForDownload: TrackForDownload[]
-  collection?: string
+  labelText?: string
+  collectionId?: number
+  isFavoritesDownlaod?: boolean
```
Comment (Kyle-Shanks) [inline packages/mobile/src/components/offline-downloads/DownloadToggle.tsx:31]: downlaod

Comment (amendelsohn) [reply packages/mobile/src/components/offline-downloads/DownloadToggle.tsx:31]: 🤣 If I see the word download one more time 😡

File: packages/mobile/src/services/offline-downloader/offline-storage.ts [hunk 2]
```diff
@@ -61,7 +166,7 @@ export const listTracks = async (): Promise<string[]> => {
     return []
   }
   const files = await readDir(tracksDir)
-  return files.filter((file) => file.isDirectory).map((file) => file.name)
+  return files.filter((file) => file.isDirectory()).map((file) => file.name)
```
Comment (Kyle-Shanks) [inline packages/mobile/src/services/offline-downloader/offline-storage.ts:169]: nice 🔍

**Review by Kyle-Shanks — APPROVED** [review]
Good stuff. Liking the changes to collectionId too to prevent confusion 👍

**Pull request merged**